packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -65,6 +72,228 @@ export default class VrackMoveDialogCtrl {
+    this.IP_SERVICE = IP_SERVICE;
+    this.serviceIconClassMapper = serviceIconClassMapper;
+    this.coreConfig = coreConfig;
+  }
+
+  /**
+   * @name filterAllowedServicesOnly
+   * @description takes the object describing the services,
+   * and returns the object after filtering each of them only for available services.
+   * @param allServicesParam
+   * @returns {*&{dedicatedCloud: *, dedicatedServer: T[], dedicatedServerInterface: {readonly dedicatedServer?: *}[], managedBareMetal: *}}
+   */
+  static filterAllowedServicesOnly(allServicesParam) {
+    const allFamilies = Object.assign(allServicesParam, SERVICE_FAMILIES_MAP);
+
+    return {
+      ...allFamilies,
+      dedicatedCloud: allServicesParam.dedicatedCloud.filter((service) => {
+        return (
+          VrackMoveDialogCtrl.isServiceAllowed(service) &&
+          service.productReference === 'EPCC'
+        );
+      }),
+      managedBareMetal: allServicesParam.dedicatedCloud.filter((service) => {
+        return (
+          VrackMoveDialogCtrl.isServiceAllowed(service) &&
+          service.productReference === 'MBM'
+        );
+      }),
+      dedicatedServer: allServicesParam.dedicatedServer.filter(
+        VrackMoveDialogCtrl.isServiceAllowed,
+      ),
+      dedicatedServerInterface: allServicesParam.dedicatedServerInterface.filter(
+        ({ dedicatedServer }) =>
+          VrackMoveDialogCtrl.isServiceAllowed(dedicatedServer),
+      ),
+    };
+  }
+
+  static isServiceAllowed(service) {
+    return (
+      !service.expired &&
+      [STATUS.ok, STATUS.delivered].includes(service.state) &&
+      service.status === STATUS.ok
+    );
+  }
+
+  static hasServices(services) {
+    return keys(services).length > 0;
+  }
+
+  /**
+   * make a niceName with the product description
+   * @param service
+   */
+  static getOvhCloudConnectNiceName(service) {
+    return {
+      id: service.uuid,
+      niceName: service.product,
+      trueServiceType: 'ovhCloudConnect',
+    };
+  }
+
+  static getDedicatedServerNiceName(service) {
+    const formattedService = Object.create(service);
+    formattedService.id = service.name;
+    // by default the reverse seem to be equal to the name,
+    // so do not display redundant information.
+    if (service.reverse && service.reverse !== service.name) {
+      formattedService.niceName = `${service.name} (${service.reverse})`;
+    } else {
+      formattedService.niceName = service.name;
+    }
+    formattedService.trueServiceType = 'dedicatedServer';
+    return formattedService;
+  }
+
+  static getDedicatedServerInterfaceNiceName(service) {
+    const formattedService = VrackMoveDialogCtrl.getDedicatedServerNiceName(
+      service.dedicatedServer,
+    );
+    formattedService.id = service.dedicatedServerInterface;
+    formattedService.niceName = `${formattedService.niceName} - ${service.name}`;
+    formattedService.trueServiceType = 'dedicatedServerInterface';
```

Comment:
You should move in constant file "dedicatedServerInterface". the same for all constant for trueServiceType

Reply:
updated to constant